udata_front/theme/gouvfr/assets/js/components/MarkAsNoSpam/MarkAsNoSpam.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+  <button
+    class="fr-btn fr-btn--sm fr-btn--icon-right fr-icon-feedback-fill"
+    @click="markAsNoSpam"
+    v-if="show"
+  >
+    {{$t('Mark as no spam')}}
+  </button>
+</template>
+<script setup lang="ts">
+import { computed, defineProps } from "vue";
+import { user } from "../../config";
+import { api } from "../../plugins/api";
+
+const props = defineProps<{
+  url: string,
+  spam: {
+    status: string
+  }
+}>();
+
+const markAsNoSpam = () => {
+  return api
+    .delete(props.url)
+    .then(() => {
+      window.location.reload();
+    });
+};
+
+const show = computed(() => props.spam
+  && props.spam.status
+  && props.spam.status === 'potential_spam'
+  && user
```

Comment:
```suggestion
```
user is always defined.

Reply:
Even if nobody is connected? What's inside in this case?

Reply:
The `user` object is always defined in the config file with empty values for each field : https://github.com/etalab/udata-front/blob/master/udata_front/theme/gouvfr/assets/js/config.ts#L48